ecl/hql/hqlattr.cpp

```diff
@@ -4003,8 +4002,11 @@ void CHqlDataset::addProperty(ExprPropKind kind, IInterface * value)
     {
         SpinBlock block(*propertyLock);
```

Comment:
And this one?

Reply:
I agree

Reply:
ok above, but not sure it is really needed in this block ?

Reply:
@ghalliday I don't think you answered this question?

Reply:
Same reason as above - it could use compare and swap to avoid the lock, but needs something otherwise two threads setting it at the same time would leak one of the items added (or crash if it was released).